icegatherer.go

```diff
@@ -185,7 +187,9 @@ func (g *ICEGatherer) Close() error {
 	}
 
 	g.agent = nil
-	g.setState(ICEGathererStateClosed)
+	setStateClose = func() {
+		g.setState(ICEGathererStateClosed)
```

Comment:
Oh I get what you are running up against. You can only call `setState` when you don't have the lock.

I would make it an atomic as well, then you don't have to worry about juggling mutexes as well. State already has a getter so should be an easy change!

Reply:
I'll also make state atomic in this PR.